index.js

```diff
@@ -59,6 +86,7 @@ module.exports = function(fn) {
 
     res.end = function(chunk, encoding, cb) {
       debug('end called');
+      endCalled = true;
```

Comment:
In the overwritten res.writeHead method, I'll do one of the following things:

1) If we're not supposed to be intercepting, I'll call originalWriteHead.
2) If end has been called already, I'll call originalWriteHead. (this will most likely result in an error writing headers after first chunk, but I'm just obeying orders from whoever is using me).
3) If we should intercept, and end haven't been called, I'll unpack the res.writeHeaders call, by calling `res.status` and `res.set`.

Due to the unpacking in option 3, I wont have to call writeHead again. The reason that I do this is first and foremost that it's not really the idiomatic express-way to use res.writeHead directly. And secondly, it would be quite hard and error prone to attempt to cache the values and call them again later. If what we are intercepting is setting a header with res.writeHead, and we were just calling res.writeHead with the same values later, we would not be able to overwrite or alter that header for that request. Unpacking it and setting the values independently with res.set allows us to do just that.

The only drawback to this approach is that I cannot figure out how to support the custom status messages, but that seemed like an OK compromise to me.